create-directories.bash

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+# Given a source directory, /source, and a target directory,
+# /target/foo/bar/bazz, we want to "clone" the target structure
+# from source into the target. Essentially, we want both
+# /source/target/foo/bar/bazz and /target/foo/bar/bazz to exist
+# on the filesystem. More concretely, we'd like to map
+# /state/etc/ssh/example.key to /etc/ssh/example.key
+#
+# To achieve this, we split the target's path into parts -- target, foo,
+# bar, bazz -- and iterate over them while accumulating the path
+# (/target/, /target/foo/, /target/foo/bar, and so on); then, for each of
+# these increasingly qualified paths we:
+#
+#   1. Ensure both /source/qualifiedPath and qualifiedPath exist
+#   2. Copy the ownership of the source path into the target path
+#   3. Copy the mode of the source path into the target path
+
+# Get inputs from command line arguments
+sourceBase="$1"
+target="$2"
+
+# trim trailing slashes the root of all evil
+sourceBase="${sourceBase%/}"
+target="${target%/}"
+
+# check that the source exists, if it doesn't exit the underlying
+# scope (the activation script will continue)
```

Comment:
This comment seems out-of-date?

Reply:
Yeah, it should be up-to-date now. I also added a note for the user to check permissions on the auto-created directories and made it print to `stderr` - while not strictly an error, this makes it easier to find in the log.